decidim-proposals/app/permissions/decidim/proposals/permissions.rb

```diff
@@ -103,6 +111,42 @@ def can_unvote_proposal?
 
         toggle_allow(is_allowed)
       end
+
+      def apply_collaborative_draft_permissions(permission_action)
+        case permission_action.action
+        when :create
+          can_create_collaborative_draft?
+        when :edit
+          can_edit_collaborative_draft?
+        when :publish
+          can_publish_collaborative_draft?
+        when :request_access
+          can_request_access_collaborative_draft?
+        end
+      end
+
+      def collaborative_draft
+        @collaborative_draft ||= context.fetch(:collaborative_draft, nil)
+      end
+
+      def can_create_collaborative_draft?
+        toggle_allow(authorized?(:create) && current_settings&.creation_enabled?)
```

Comment:
Shouldn't you use `collaborative_drafts_enabled?`?

Reply:
maybe better check both of them, as `collaborative_drafts_enabled` can be enabled but `creation_disabled`, do't you think so?

Reply:
I'll also add `collaborative_drafts_enabled` to the below `can_edit_collaborative_draft?`, `can_publish_collaborative_draft?` and `can_request_access_collaborative_draft?` as these actions can only be performed if `collaborative_drafts_enabled`, ok?